src/main/java/org/opensearch/ad/util/RestHandlerUtils.java

```diff
@@ -143,6 +155,21 @@ private static String validateFeatures(List<Feature> features) {
         return errorMsgBuilder.toString();
     }
 
+    public static boolean isExceptionCausedByInvalidQuery(Exception ex) {
+        if (!(ex instanceof SearchPhaseExecutionException)) {
+            return false;
+        }
+        SearchPhaseExecutionException exception = (SearchPhaseExecutionException) ex;
+        // If all shards return bad request and failure cause is IllegalArgumentException, we
```

Comment:
Minor, "If all shards..." should be "If any shard.." ?

Reply:
changed